core/templates/services/playthrough.service.ts

```diff
@@ -16,330 +16,350 @@
-
-    var misTracker = false;
-    var cstTracker = false;
+    visitedStates = [];
+    misTracker: boolean = false;
+    cstTracker: boolean = false;
 
-    var removeOldQuitAction = function() {
-      var quitAction = playthrough.actions[playthrough.actions.length - 1];
+    private _removeOldQuitAction(): void {
+      var quitAction = this.playthrough.actions[
+        this.playthrough.actions.length - 1];
       // After the second quit action is recorded, the first quit is removed
       // using this method. This ensures that there are only two quit actions
       // in the playthrough actions list at a time.
-      playthrough.actions = playthrough.actions.filter(function(action) {
-        return action.actionType !== ACTION_TYPE_EXPLORATION_QUIT;
-      });
-      playthrough.actions.push(quitAction);
-    };
+      this.playthrough.actions = this.playthrough.actions.filter(
+        function(action) {
+          return (
+            action.actionType !== AppConstants.ACTION_TYPE_EXPLORATION_QUIT);
+        });
+      this.playthrough.actions.push(quitAction);
+    }
 
-    var _determineIfLearnerIsInSamplePopulation = function(probability) {
+    private _determineIfLearnerIsInSamplePopulation(
+        probability: number): boolean {
       return Math.random() < probability;
-    };
+    }
 
-    var createMultipleIncorrectIssueTracker = function(initStateName) {
-      if (misTracker) {
+    private _createMultipleIncorrectIssueTracker(initStateName: string) {
+      if (this.misTracker) {
         return;
       }
-      multipleIncorrectStateName = {
+      this.multipleIncorrectStateName = {
         state_name: initStateName,
         num_times_incorrect: 0
       };
-      misTracker = true;
-    };
+      this.misTracker = true;
+    }
 
-    var createCyclicIssueTracker = function(initStateName) {
-      if (cstTracker) {
+    private _createCyclicIssueTracker(initStateName: string) {
+      if (this.cstTracker) {
         return;
       }
-      cycleIdentifier = {
+      this.cycleIdentifier = {
         cycle: '',
         num_cycles: 0
       };
-      visitedStates.unshift(initStateName);
-      cstTracker = true;
-    };
+      this.visitedStates.unshift(initStateName);
+      this.cstTracker = true;
+    }
 
-    var incrementIncorrectAnswerInMultipleIncorrectIssueTracker = function() {
-      multipleIncorrectStateName.num_times_incorrect += 1;
-    };
+    private _incrementIncorrectAnswerInMultipleIncorrectIssueTracker(): void {
+      this.multipleIncorrectStateName.num_times_incorrect += 1;
+    }
 
-    var recordStateTransitionInMultipleIncorrectIssueTracker = function(
-        destStateName) {
-      if (multipleIncorrectStateName.num_times_incorrect <
-        NUM_INCORRECT_ANSWERS_THRESHOLD) {
-        multipleIncorrectStateName.state_name = destStateName;
-        multipleIncorrectStateName.num_times_incorrect = 0;
+    private _recordStateTransitionInMultipleIncorrectIssueTracker(
```

Comment:
Missing return type, please add the missing types everywhere.

Reply:
Done.